File: Javascript/dfs/네트워크.js
function solution(n, computers) {
	let answer = 0;
	let nn = computers.length;
	let m = computers[0].length;
	let visited = Array.from({ length: n }, () => false);

	function DFS(x) {
		visited[x] = true;

		for (let i = 0; i < m; i++) {
			if (computers[x][i] === 1 && !visited[i]) {
				DFS(i);
			}
		}
	}

	for (let i = 0; i < nn; i++) {
		if (!visited[i]) {
			DFS(i);
			answer += 1;
		}
	}
	return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [False] * n
    
    def dfs(x):
        visited[x] = True
        
        if x == n:
            return
        
        for i in range(n):
            if not visited[i] and computers[x][i] == 1:
                dfs(i)
        
    for i in range(len(computers)):
        if not visited[i] and computers[i][i] == 1:
            dfs(i)
            answer += 1
            
    return answer